src/fromager/bootstrapper.py

```diff
@@ -316,11 +303,81 @@ def _handle_prebuilt_sources(
             logger.info(f"{req.name}: updating temporary mirror with pre-built wheel")
             shutil.copy(wheel_filename, dest_name)
             server.update_wheel_mirror(self.ctx)
-        unpack_dir = self.ctx.work_dir / f"{req.name}-{resolved_version}"
-        if not unpack_dir.exists():
-            unpack_dir.mkdir()
+        unpack_dir = self._create_unpack_dir(req, resolved_version)
         return (resolved_version, wheel_url, wheel_filename, unpack_dir)
 
+    def _download_wheel_from_cache(
+        self, req: Requirement, resolved_version: Version
+    ) -> tuple[pathlib.Path | None, pathlib.Path | None]:
+        if not self.cache_wheel_server_url:
+            return None, None
+        logger.info(f"{req.name}: checking if wheel was already built")
+        try:
+            wheel_url, _ = resolver.resolve(
+                ctx=self.ctx,
+                req=Requirement(f"{req.name}=={resolved_version}"),
+                sdist_server_url=self.cache_wheel_server_url,
+                include_sdists=False,
+                include_wheels=True,
+            )
+            cached_wheel = wheels.download_wheel(
+                req=req, wheel_url=wheel_url, output_directory=self.ctx.wheels_downloads
+            )
+            server.update_wheel_mirror(self.ctx)
+
+            pbi = self.ctx.package_build_info(req)
+            expected_build_tag = pbi.build_tag(resolved_version)
+            dist_name, dist_version, build_tag, _ = wheels.extract_info_from_wheel_file(
```

Comment:
Do we have to download the wheel because we want to call this function and it needs the file? Or can we parse the URL to get the build tag and avoid doing the download if it's wrong?

Reply:
Ah yes I think there is a way, let me try

Reply:
done